Tests/Tests/STPAddressTests.m

```diff
@@ -458,6 +503,81 @@ - (void)testContainsRequiredShippingAddressFields {
     XCTAssertTrue([address containsRequiredShippingAddressFields:PKAddressFieldAll]);
 }
 
+- (void)testContainsContentForShippingAddressFields {
+    STPAddress *address = [STPAddress new];
+
+    // Empty address should return false for everything
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldNone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldName]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPhone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldEmail]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPostalAddress]);
+
+    // Name
+    address.name = @"Smith";
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldNone]);
+    XCTAssertTrue([address containsContentForShippingAddressFields:PKAddressFieldName]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPhone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldEmail]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPostalAddress]);
+    address.name = @"";
+
+    // Phone
+    address.phone = @"1";
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldNone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldName]);
+    XCTAssertTrue([address containsContentForShippingAddressFields:PKAddressFieldPhone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldEmail]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPostalAddress]);
+    address.phone = @"";
+
+    // Email
+    address.email = @"f";
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldNone]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldName]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPhone]);
+    XCTAssertTrue([address containsContentForShippingAddressFields:PKAddressFieldEmail]);
+    XCTAssertFalse([address containsContentForShippingAddressFields:PKAddressFieldPostalAddress]);
+    address.email = @"";
+
+    // Test every property that contributes to the full address
+    // This is *not* refactoring-safe, but I think it's better than a bunch more duplicated code
+    for (NSString *propertyName in @[@"line1", @"line2", @"city", @"state", @"postalCode", @"country"]) {
```

Comment:
Maybe do something like `NSStringFromSelector(@selector(line1))` instead of `@"line1"` for some more compiler safety?